lib/python/rose/apps/ana_builtin/grepper.py

```diff
@@ -41,20 +45,34 @@ def run_command_and_check(self):
         self.command = self.command.format(*self.files)
         returncode, stdout, stderr = self.run_command(self.command)
         if returncode == 0:
-            self.parent.reporter(stdout, prefix="[INFO] ")
+            self.reporter(stdout, prefix="[INFO] ")
             self.passed = True
         else:
-            self.parent.reporter(stderr, prefix="[FAIL] ")
+            self.reporter(stderr, prefix="[FAIL] ")
+
+    def check_for_skip(self):
+        # If the users config options specified that the task should be
+        # ignored if all of its files were missing, set skipped attribute here
+        if self.skip_if_missing and len(self.files) > 0:
+            if all([not os.path.exists(fname) for fname in self.files]):
```

Comment:
I'm fairly sure Python _isn't_ clever enough for that - the comprehension will get evaluated in full and then the `any` will only scan the result as far as it needs to find a `True`...

However I think I'll still make this change because your version reads better